homeassistant/components/media_player/__init__.py

```diff
@@ -639,6 +656,14 @@ async def async_set_shuffle(self, shuffle):
         """Enable/disable shuffle mode."""
         await self.hass.async_add_job(self.set_shuffle, shuffle)
 
+    def set_repeat(self, repeat):
+        """Set repeat mode."""
+        raise NotImplementedError()
+
+    async def async_set_repeat(self, repeat):
+        """Set repeat mode."""
+        await self.hass.async_add_job(self.set_repeat, repeat)
```

Comment:
We should use `hass.async_add_executor_job`. `hass.async_add_job` is legacy for general use.